bluemira/geometry/base.py

```diff
@@ -122,14 +122,23 @@ def _converter(func):
     def _check_boundary(self, objs):
         """
         Check if objects objs can be used as boundaries.
+
+        Note: empty BluemiraGeo are allowed in case of objs == None.
         """
+        if objs is None:
+            return objs
+
         if not hasattr(objs, "__len__"):
             objs = [objs]
+
         check = False
         for c in self._boundary_classes:
-            check = check or (all(isinstance(o, c) for o in objs))
+            # # in case of obj = [], this check returns True instead of False
+            # check = check or (all(isinstance(o, c) for o in objs))
+            for o in objs:
+                check = check or isinstance(o, c)
             if check:
-                return objs
+                return tuple(objs)
```

Comment:
@CoronelBuendia If it does not disturb the fact that self._boundary is a list and self.boundary returns a tuple, I would propose converting the list into a tuple in the boundary getter (in this way it is still allowed to do self._boundary.append(things) as in some file).